ui/js/page/discover/view.jsx

```diff
@@ -37,10 +38,31 @@ const FeaturedCategory = props => {
 class DiscoverPage extends React.PureComponent {
   componentWillMount() {
     this.props.fetchFeaturedUris();
+    this.scrollListener = this.handleScroll.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.props.isNavigatingBack) {
+      const scrollY = parseInt(lbry.getClientSetting("prefs_scrolly"));
+      if (!isNaN(scrollY)) {
+        const restoreScrollPosition = () => {
+          window.scrollTo(0, scrollY);
+        };
+        setTimeout(restoreScrollPosition, 100);
+      }
+
+      this.props.finishedNavigatingBack();
```

Comment:
@kauffj  This delegates to `doHistoryBackCompleted()`. I called it at this point because the `popstate` event was not being fired after clicking the back button. Also, `HISTORY_BACK` occurs before `CHANGE_PATH`, and this happens before `componentDidMount`. I needed a way to reliably detect that the user navigated using the back button and being able to check the flag when the component mounts was how I could do it (although I figure there might be a better way but I'm not seeing it at this point).